src/scribe_data/check/check_language_metadata.py

```diff
@@ -0,0 +1,213 @@
+        else:
+            # If the parent exists, only check for missing sub-languages
+            ref_sub_languages = reference_languages[lang].get("sub_languages", {})
+
+            if "sub_languages" in details:
+                for sub_lang in details["sub_languages"]:
+                    if sub_lang not in ref_sub_languages:
+                        missing_languages.append(f"{lang}/{sub_lang}")
+
+    return missing_languages
+
+
+def validate_language_properties(languages_dict: dict) -> dict:
+    """
+    Validates the presence of 'qid' and 'iso' properties for each language and its sub-languages.
+
+    Parameters
+    ----------
+        languages_dict (dict): A dictionary where each key is a language, and the value is another dictionary containing details about the language. If the language has sub-languages, they are stored under the 'sub_languages' key.
+
+    Returns
+    -------
+        dict: A dictionary with two lists:
+              - "missing_qids": Languages or sub-languages missing the 'qid' property.
+              - "missing_isos": Languages or sub-languages missing the 'iso' property.
+
+              Each entry in these lists is in the format "parent_language - sub_language" for sub-languages,
+              or simply "parent_language" for the parent languages.
+    """
+    missing_qids = []
+    missing_isos = []
+
+    for lang, details in languages_dict.items():
+        # Check if the language has sub-languages
+        if "sub_languages" in details:
+            sub_languages = details["sub_languages"]
+
+            # Validate each sub-language
+            for sub_lang, sub_details in sub_languages.items():
+                if "qid" not in sub_details:
+                    missing_qids.append(f"{lang}/{sub_lang}")
+                if "iso" not in sub_details:
+                    missing_isos.append(f"{lang}/{sub_lang}")
+        else:
+            # Validate the parent language itself
+            if "qid" not in details:
+                missing_qids.append(lang)
+            if "iso" not in details:
+                missing_isos.append(lang)
+
+    return {"missing_qids": missing_qids, "missing_isos": missing_isos}
+
+
+def check_language_metadata():
+    """
+    Validates language metadata by performing the following checks:
+
+    1. Ensures that all languages listed in `language_data_extraction` are present in `language_metadata.json`, and vice versa.
+    2. Checks if each language in `language_metadata.json` has the required properties:
+       - 'qid' (a unique identifier)
+       - 'iso' (ISO language code)
+
+    This function helps identify missing languages or missing properties, ensuring data consistency across both sources.
+
+    Raises:
+    -------
+        SystemExit:
+            If any missing languages or properties are found, the function exits the script with a status code of 1.
+    """
+    languages_in_metadata = {key.lower(): value for key, value in _languages.items()}
+
```

Comment:
Does this line  
`languages_in_metadata = {key.lower(): value for key, value in _languages.items()}`  
give all the languages in `_languages`, including the sublanguages?

Reply:
yes. but it doesn't lowercase all the sublanguages if any is mistakenly written, I just noticed.

Reply:
There is also a helper function called `list_all_languages` at the bottom of `utils.py` that lists all the languages in the JSON file. You can make use of it here to check for sublanguages; it would be helpful.

Reply:
Thanks for the suggestion! Could you explain further how best to use it in this context? I want to include parent languages so I convert the languages directory to match the format of `language_metadata.json` for easier comparison. Since `list_all_languages` doesn’t show the parent languages, I’m unsure how to use it effectively for my comparison.